app/Actions/Photo/Create.php

```diff
@@ -68,30 +68,47 @@ public function add(NativeLocalFile $source_file, ?AbstractAlbum $album, ?int $f
 	{
 		$this->checkQuota($source_file);
 
+		/** @var InitDTO $init_dto */
 		$init_dto = new InitDTO(
 			parameters: $this->strategy_parameters,
 			source_file: $source_file,
 			album: $album,
 			file_last_modified_time: $file_last_modified_time
 		);
 
-		/** @var InitDTO $init_dto */
+		$pre_pipes = [
+			Init\AssertSupportedMedia::class,
+			Init\FetchLastModifiedTime::class,
+			Init\LoadFileMetadata::class,
+			Init\FindDuplicate::class,
+		];
+
+		$post_pipes = [
+			Init\InitParentAlbum::class,
+			Init\LoadFileMetadata::class,
+			Init\FindLivePartner::class,
+		];
+
+		if ($this->strategy_parameters->import_mode->shall_resync_metadata) {
```

Comment:
I am not so keen of this conditional removal, I would rather see a Pipe that execute depending of the shall_resync_metadata that could be added as an attribute in InitDTO.

This can also be inheriting from the original LoadFileMetadata.
And LoadFileMetadata could always load if the Exif data are null.